File: verify_no_emojis.js
const fs = require('fs');
const path = require('path');

const PATTERNS = [
    /\$\{.*emoji\s*\|\|\s*'❓'\}/,
    /this\.nextElementSibling\.style\.display\s*=\s*'block'/,
    /onerror=.*p\.emoji/,
    /onerror=.*pokemon\.emoji/,
    /battle-sprite-emoji/,
    /player-sprite-emoji/,
    /enemy-sprite-emoji/,
    /team-emoji-/
];

const EXCLUDE_FILES = [
    '01_data.js',
    '23_market.js'
];

function verifyFiles(dir) {
    let issues = [];
    const files = fs.readdirSync(dir);

    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            issues = issues.concat(verifyFiles(fullPath));
            continue;
        }

        if (!file.endsWith('.js') || EXCLUDE_FILES.includes(file)) continue;

        const content = fs.readFileSync(fullPath, 'utf-8');
        const lines = content.split('\n');

        lines.forEach((line, i) => {
            for (const pattern of PATTERNS) {
                if (pattern.test(line)) {
                    // Ignore legitimate item/icon fallbacks
                    if (line.includes('item.icon') || line.includes('item.sprite') || line.includes('stoneIcon')) continue;
                    issues.push(`[${file}:${i + 1}] ${line.trim()}`);
                }
            }
        });
    }
    return issues;
}

const jsDir = path.join(__dirname, 'js');
const issues = verifyFiles(jsDir);

if (issues.length > 0) {
    console.log(`Found ${issues.length} potential emoji placeholders:`);
    issues.forEach(issue => console.log(issue));
    process.exit(1);
} else {
    console.log("Success: No Pokemon emoji placeholders found in JS files.");
    process.exit(0);
}

File: scripts/verify_no_emojis.py
import os
import re

# Define patterns that likely represent emoji placeholders for Pokemon sprites
PATTERNS = [
    r"\$\{.*emoji\s*\|\|\s*'❓'\}",
    r"this\.nextElementSibling\.style\.display\s*=\s*'block'",
    r"onerror=.*p\.emoji",
    r"onerror=.*pokemon\.emoji",
    r"battle-sprite-emoji",
    r"player-sprite-emoji",
    r"enemy-sprite-emoji",
    r"team-emoji-",
]

# Files to exclude from verification (e.g., constants, item shops, or non-sprite logic)
EXCLUDE_FILES = [
    "01_data.js",  # Contains the actual database with emoji definitions
    "23_market.js", # Item/Type icons
]

def verify_files(directory):
    found_issues = []
    
    for root, _, files in os.walk(directory):
        for file in files:
            if not file.endswith('.js') or file in EXCLUDE_FILES:
                continue
                
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                lines = f.readlines()
                for i, line in enumerate(lines):
                    for pattern in PATTERNS:
                        if re.search(pattern, line):
                            # Special case: allow reputation shop icons (items)
                            if "item.icon" in line or "item.sprite" in line:
                                continue
                            if "stoneIcon" in line:
                                continue
                            found_issues.append((file, i + 1, line.strip()))
                            
    return found_issues

if __name__ == "__main__":
    js_dir = "d:\\Documentos\\GitHub\\PokeBorrador\\js\\"
    issues = verify_files(js_dir)
    
    if issues:
        print(f"Found {len(issues)} potential emoji placeholders:")
        for file, line_num, content in issues:
            print(f"[{file}:{line_num}] {content}")
        exit(1)
    else:
        print("Success: No Pokemon emoji placeholders found in JS files.")
        exit(0)
